kotlinx-coroutines-core/common/src/Await.kt

```diff
@@ -7,6 +7,14 @@ package kotlinx.coroutines
 import kotlinx.atomicfu.*
 import kotlin.coroutines.*
 
+/**
+ * Suspends until cancellation, in which case it will throw a [CancellationException].
+ *
+ * Handy because it returns [Nothing], allowing it to be used in any coroutine,
```

Comment:
Actually, it would be great to show some small practical example here, if you have one in mind.

Reply:
Added two examples. I failed to find simpler examples for now, but it's not public API so it can be changed in the future.

Also, I know `trySend` doesn't exist yet, but I thought it would look nicer than `runCatching { offer(…) }` or a try/catch.